z3/src/tactic.rs

```diff
@@ -4,43 +4,25 @@ use std::result::Result;
 use std::str::Utf8Error;
 
 use z3_sys::*;
+use ApplyResult;
 use Context;
 use Goal;
 use Probe;
 use Params;
 use Tactic;
 use Z3_MUTEX;
 
-#[derive(Clone, Debug)]
-pub struct ApplyResult<'ctx> {
-    ctx: &'ctx Context,
-    z3_apply_result: Z3_apply_result,
-}
-
 impl<'ctx> ApplyResult<'ctx> {
-    pub fn list_subgoals(&self) -> impl Iterator<Item = Goal> {
-        unsafe {
-            let num_subgoals = Z3_apply_result_get_num_subgoals(self.ctx.z3_ctx, self.z3_apply_result);
-            (0..num_subgoals).into_iter().map(move |i| {
-                let guard = Z3_MUTEX.lock().unwrap();
-                let sg = Z3_apply_result_get_subgoal(self.ctx.z3_ctx, self.z3_apply_result, i);
-                Z3_goal_inc_ref(self.ctx.z3_ctx, sg);
-                Goal::new_from_z3_type(self.ctx, sg, true, true, true)
-            })
-        }
-    }
-}
-
-impl<'ctx> Clone for Tactic<'ctx> {
-    fn clone(&self) -> Self {
```

Comment:
I understand the point of the change and I'm fine with it, but in the future it would be better-suited to a separate PR